archive/multisource.go

```diff
@@ -87,31 +79,35 @@ func newSpanScanner(ctx context.Context, ark *Archive, zctx *resolver.Context, f
 // Otherwise, the sources come from localizing the given alternative paths to
 // each chunk in the archive, recognizing "_" as the chunk file itself, with no
 // defined ordering.
-func NewMultiSource(ark *Archive, altPaths []string) driver.MultiSource {
+func NewMultiSource(ark *Archive, altPaths []string) Source {
 	if len(altPaths) == 1 && altPaths[0] == "_" {
 		altPaths = nil
 	}
 	if len(altPaths) != 0 {
 		return &chunkMultiSource{
 			ark:      ark,
 			altPaths: altPaths,
+			stats:    &ChunkStats{},
 		}
 	}
-	return &spanMultiSource{ark}
+	return &spanMultiSource{ark, &ChunkStats{}}
 }
 
 type spanMultiSource struct {
-	ark *Archive
+	ark   *Archive
+	stats *ChunkStats
 }
 
+var _ Source = &spanMultiSource{}
```

Comment:
This is redundant; NewMultiSource wouldn't compile if spanMultiSource didn't implement Source.

Reply:
Good point.